packages/x-date-pickers/src/internals/components/CalendarOrClockPicker/useFocusManagement.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import { CalendarPickerView } from '../../models/views';
+
+interface FocusState {
+  focusedView: CalendarPickerView | null;
+}
+
+type FocusModificationPayload = { view: CalendarPickerView };
+
+type ReducerAction<TType, TAdditional = {}> = { type: TType } & TAdditional;
+
+export const createCalendarStateReducer =
+  () =>
+  (
+    state: FocusState,
+    action:
+      | ReducerAction<'focus', FocusModificationPayload>
```

Comment:
I'm really not a fan of the `ReducerAction` interface that I saw elsewhere on this project
I find it hard to render.

I would prefer to use basic types

```ts
type FocusManagementAction = 
  | { type: 'focus', view: CalendarPickerView } 
  | { type: 'blur', view: CalendarPickerView }
```

Or if you are confident that all actions need the view

```ts
type FocusManagementAction = { action: 'focus' | 'blur', view: CalendarPickerView }
```

Reply:
By the way, can't we just use a `useState` here ?
The reducer feels somehow overengineered

Reply:
Yes, it have been simplified with time an so does not need reducer anymore